.github/workflows/build-artifact.yaml

```diff
@@ -54,14 +79,35 @@ jobs:
       - name: Set up dynamic env vars for build
         run: |
           echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
+          echo "RUNNER_UID=$(id -u)" >> $GITHUB_ENV
+          echo "RUNNER_GID=$(id -g)" >> $GITHUB_ENV
       - name: Update submodules
         run: |
           git submodule update --init --recursive
+      - name: Generate docker tag
+        id: generate-docker-tag
+        uses: ./.github/actions/generate-docker-tag
+        with:
+          image: ${{ inputs.os }}
       - name: Build tt-metal and libs
-        run: |
-          nice -n 19 cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=${{ inputs.build-type }} -DENABLE_TRACY=${{ inputs.tracy }}
-          nice -n 19 cmake --build build --target tests
-          nice -n 19 cmake --build build --target install
+        uses: addnab/docker-run-action@v3
+        with:
+          image: ${{ env.TT_METAL_DOCKER_IMAGE_TAG }}
+          options: |
+            -u ${{ env.RUNNER_UID }}:${{ env.RUNNER_GID }}
+            -v ${{ github.workspace }}:${{ github.workspace }}
+            -v /etc/passwd:/etc/passwd:ro
```

Comment:
Give it a go to remove these mapped folders of /etc.

Reply:
@tt-rkim This is the offending line we are trying to see if it still causes issues with permissions

Reply:
research via trial and error says we need all 3 mounts for no permission issues
Tried
passwd
passwd/shadow
bashrc
passwd/shadow/bashrc
was only using all 3 can we avoid permission issues